skfda/tests/test_pace.py

```diff
@@ -0,0 +1,607 @@
+            sample_fd,
+            AttributeError,
+            "The sample size must be bigger",
+        ],
+        [
+            PACE(
+                n_components=5,
+                bandwidth_mean=np.array([0.1, 10]),
+                bandwidth_cov=np.array([0.1, 10]),
+            ),
+            FDataIrregular(
+                start_indices=[0, 1, 2, 3, 4],
+                points=[[1.0], [2.0], [3.0], [4.0], [5.0]],
+                values=[[1.0], [2.0], [3.0], [4.0], [5.0]],
+            ),
+            ValueError,
+            "Unable to perform computations with one measurement",
+        ],
+        [
+            PACE(
+                n_components=2,
+                bandwidth_mean=0.1,
+                bandwidth_cov=0.1,
+            ),
+            sample_fd,
+            ValueError,
+            "Covariance matrix has invalid eigenvalues",
+        ],
+    ],
+)
+def input_fd_raises_fixture(
+    request: Any,
+) -> tuple[PACE, FDataIrregular, type[Exception], str]:
+    """Fixture for checking incorrect input data forms."""
+    return tuple(request.param)
+
+
+@pytest.fixture
+def matlab_mean_data() -> NDArrayFloat:
+    """Fixture for getting the expected mean result from Matlab."""
+    return np.array(
+        [
+            986.1189,989.1560,991.6712,993.5323,994.6030,994.7466,993.8285,
+            991.7216,988.3114,983.5016,977.2211,969.4295,960.1220,949.3330,
+            937.1374,923.6488,909.0164,893.4185,860.1407,842.8924,825.5250,
+            808.2426,791.2328,774.6622,758.6738,743.3849,728.8873,715.2470,
+            702.5060,690.6839,679.7799,669.7756,660.6366,652.3158,644.7553,
+            637.8886,631.6432,625.9426,620.7085,615.8628,611.3299,607.0380,
+            602.9214,598.9214,594.9876,591.0794,587.1662,583.2283,579.2575,
+            575.2572,571.2423,567.2387,563.2831,559.4211,555.7061,552.1970,
+            548.9562,546.0469,543.5310,541.4660,
+        ],
+    )
+
+@pytest.fixture
+def matlab_cov_data() -> NDArrayFloat:
+    """Fixture for getting the expected covariance result from Matlab."""
+    return np.array(
+        [
+            57998.8116710191,62752.6196878704,66822.5917724559,70090.6189419065,72455.8559963659,73840.9329958453,74197.8524267375,73516.4494604902,71836.8888653427,69263.9570532368,65976.0631739253,62219.3605331594,58281.5073698752,54450.3502745921,50972.7124777187,48028.3786604451,45724.2041981697,44102.3859545357,43153.2162866848,42825.7373080377,43034.709954643,43665.7234623061,44581.4908032739,45632.0259474352,46669.955740806,47569.9729497416,48248.9853074748,48681.8879541345,48908.0856330062,49026.1770415017,49177.7013568036,49523.9078610176,50220.8570988707,51397.6804714531,53141.3507007542,55489.7140012399,58433.0959194021,61923.4118515136,65888.4312679218,70248.0552453998,74929.5666612005,79879.7397211635,85072.9261296176,90515.2193645554,96245.3174237278,102332.838211533,108874.742709948,115990.289422891,123814.650458266,132491.050560917,142161.134174194,
```

Comment:
You are comparing this with absolute tolerance of 0.1. Apart from the fact that relative tolerance is usually preferred, it does not make sense to store all decimals then. Given all the variability that can arise from the preprocessing (smoothing) and the quadrature used, I would say that at most 3 decimals are significative, and probably less than that.